v0/kissmetrics/transform.js

```diff
@@ -13,132 +7,116 @@ const { ENDPOINT } = require("./config");
+  object.forEach(key => {
+    if (object.hasOwnProperty(key)) {
+      const value = object[key];
+      const isarray = opts.safe && is.array(value);
+      const type = Object.prototype.toString.call(value);
+      const isobject = type === "[object Object]" || type === "[object Array]";
+      const arr = [];
+
+      const newKey = opts ? opts + delimiter + key : key;
+
+      if (!opts.maxDepth) {
+        maxDepth = currentDepth + 1;
+      }
 
-        if (!isarray && isobject && arr.length && currentDepth < maxDepth) {
-          ++currentDepth;
-          return step(value, newKey);
+      value.forEach(k => {
+        if (value.hasOwnProperty(k)) {
+          arr.push(k);
         }
+      });
 
-        output[newKey] = value;
+      if (!isarray && isobject && arr.length && currentDepth < maxDepth) {
+        currentDepth += 1;
+        return step(value, newKey);
       }
+
+      output[newKey] = value;
     }
-  }
+  });
+}
 
-  step(target);
+// source : https://github.com/segment-integrations/analytics.js-integration-kissmetrics/blob/master/lib/index.js
+function flatten(target, opts) {
+  opts = opts || {};
+
+  const output = {};
+
+  step(target, opts);
 
   return output;
 }
 
 // source : https://github.com/segment-integrations/analytics.js-integration-kissmetrics/blob/master/lib/index.js
 function clean(obj) {
-  var ret = {};
+  let ret = {};
 
-  for (var k in obj) {
+  obj.forEach(k => {
     if (obj.hasOwnProperty(k)) {
-      var value = obj[k];
-      if (value === null || typeof value === "undefined") continue;
-
-      // convert date to unix
-      if (is.date(value)) {
-        ret[k] = toUnixTimestamp(value);
-        continue;
-      }
-
-      // leave boolean as is
-      if (is.bool(value)) {
-        ret[k] = value;
-        continue;
-      }
-
-      // leave  numbers as is
-      if (is.number(value)) {
-        ret[k] = value;
-        continue;
-      }
-
-      // convert non objects to strings
-      // console.log(value.toString());
-      if (value.toString() !== "[object Object]") {
-        ret[k] = value.toString();
-        continue;
-      }
-
-      // json
-      // must flatten including the name of the original trait/property
-      var nestedObj = {};
-      nestedObj[k] = value;
-      var flattenedObj = flatten(nestedObj, { safe: true });
-
-      // stringify arrays inside nested object to be consistent with top level behavior of arrays
-      for (var key in flattenedObj) {
-        if (is.array(flattenedObj[key])) {
-          flattenedObj[key] = flattenedObj[key].toString();
+      const value = obj[k];
+      if (value !== null && typeof value !== "undefined") {
+        // convert date to unix
+        if (is.date(value)) {
+          ret[k] = toUnixTimestamp(value);
+        } else if (is.bool(value)) {
+          ret[k] = value;
+        } else if (is.number(value)) {
+          ret[k] = value;
+        } else if (value.toString() !== "[object Object]") {
```

Comment:
I think, the conditions are not mutually exclusive.
so, may need to refer earlier code and keep multiple if blocks